src/module.c

```diff
@@ -3669,82 +3671,104 @@ robj *moduleTypeDupOrReply(client *c, robj *fromkey, robj *tokey, robj *value) {
+ *     version, like in the following example:
+ *
+ *         RedisModuleTypeMethods tm = {
+ *             .version = REDISMODULE_TYPE_METHOD_VERSION,
+ *             .rdb_load = myType_RDBLoadCallBack,
+ *             .rdb_save = myType_RDBSaveCallBack,
+ *             .aof_rewrite = myType_AOFRewriteCallBack,
+ *             .free = myType_FreeCallBack,
+ *
+ *             // Optional fields
+ *             .digest = myType_DigestCallBack,
+ *             .mem_usage = myType_MemUsageCallBack,
+ *             .aux_load = myType_AuxRDBLoadCallBack,
+ *             .aux_save = myType_AuxRDBSaveCallBack,
+ *             .free_effort = myType_FreeEffortCallBack,
+ *             .unlink = myType_UnlinkCallBack,
+ *             .copy = myType_CopyCallback,
+ *             .defrag = myType_DefragCallback
+ *         }
  *
  * * **rdb_load**: A callback function pointer that loads data from RDB files.
+ *
  * * **rdb_save**: A callback function pointer that saves data to RDB files.
+ *
  * * **aof_rewrite**: A callback function pointer that rewrites data as commands.
+ *
  * * **digest**: A callback function pointer that is used for `DEBUG DIGEST`.
+ *
  * * **free**: A callback function pointer that can free a type value.
- * * **aux_save**: A callback function pointer that saves out of keyspace data to RDB files.
- *   'when' argument is either REDISMODULE_AUX_BEFORE_RDB or REDISMODULE_AUX_AFTER_RDB.
- * * **aux_load**: A callback function pointer that loads out of keyspace data from RDB files.
- *   Similar to aux_save, returns REDISMODULE_OK on success, and ERR otherwise.
- * * **free_effort**: A callback function pointer that used to determine whether the module's
- *   memory needs to be lazy reclaimed. The module should return the complexity involved by
- *   freeing the value. for example: how many pointers are gonna be freed. Note that if it 
- *   returns 0, we'll always do an async free.
- * * **unlink**: A callback function pointer that used to notifies the module that the key has 
- *   been removed from the DB by redis, and may soon be freed by a background thread. Note that 
- *   it won't be called on FLUSHALL/FLUSHDB (both sync and async), and the module can use the 
- *   RedisModuleEvent_FlushDB to hook into that.
- * * **copy**: A callback function pointer that is used to make a copy of the specified key.
- *   The module is expected to perform a deep copy of the specified value and return it.
- *   In addition, hints about the names of the source and destination keys is provided.
- *   A NULL return value is considered an error and the copy operation fails.
- *   Note: if the target key exists and is being overwritten, the copy callback will be
- *   called first, followed by a free callback to the value that is being replaced.
+ *
+ * * **aux_save**: A callback function pointer that saves out of keyspace data
+ *     to RDB files. The 'when' argument is either `REDISMODULE_AUX_BEFORE_RDB`
+ *     or `REDISMODULE_AUX_AFTER_RDB`.
```

Comment:
why are these changes needed? this portion in redis.io seems fine, isn't it?

Reply:
The error on redis.io is in the bullet `typemethods_ptr` a few bullets above this one. It contains some code block `RedisModuleTypeMethods tm = { .version = REDIS....` which isn't rendered correctly.

Assuming the rendering is based on the original Markdown from https://daringfireball.net/projects/markdown/syntax#list which is used, it dictates that "List items may consist of multiple paragraphs. Each subsequent paragraph in a list item must be indented by either 4 spaces or one tab". In Github (and probably other implementations) 2 spaces indentation is enough within list items.

I changed all the bullets throughout the list to follow this, for consistency, but indeed it would be enough to do it only in the bullets with sublists and code blocks inside. I can revert these to minimize the diff instead.